src/transformers/models/rt_detr/configuration_rt_detr.py

```diff
@@ -0,0 +1,343 @@
+# coding=utf-8
+# Copyright 2024 Facebook AI Research and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" RT-DETR model configuration"""
+
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+from ..auto import CONFIG_MAPPING
+
+
+logger = logging.get_logger(__name__)
+
+RT_DETR_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "sbchoi/rtdetr_r50vd": "https://huggingface.co/sbchoi/rtdetr_r50vd/blob/main/config.json",
+}
+
+
+class RTDetrConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`RTDetrModel`]. It is used to instantiate a
+    RT_DETR model according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the RT_DETR
+    [checkpoing/todo](https://huggingface.co/checkpoing/todo) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        layer_norm_eps (`float`, *optional*, defaults to 1e-05):
+            The epsilon used by the layer normalization layers.
+        batch_norm_eps (`float`, *optional*, defaults to 1e-05):
+            The epsilon used by the batch normalization layers.
+        use_timm_backbone (`bool`, *optional*, defaults to `True`):
+            Whether the model use timm backbone.
+        backbone_config (`Union[Dict[str, Any], PretrainedConfig]`, *optional*):
+            The configuration of the backbone in a dictionary or the config object of the backbone.
+        backbone (`str`, *optional*, defaults to `"resnet50d"`):
+            Type of the backbone based on timm.
+        use_pretrained_backbone (`bool`, *optional*, defaults to `True`):
+            Whether to use pretrained weight for backbone model.
```

Comment:
The issue here is that if people want to use the RT-DETR in a completely offline setting (no internet access), the `use_pretrained_backbone=True` flag will attempt to still load the weights from the hub for the timm backbone, even though people may have downloaded an RT-DETR checkpoint from the hub.

cc @amyeroberts

Reply:
I also noticed that it is quite tricky to import resnet-50 d in this case because it has little bit different between timm and other AutoBackbone API

Reply:
> The issue here is that if people want to use the RT-DETR in a completely offline setting (no internet access), the use_pretrained_backbone=True flag will attempt to still load the weights from the hub for the timm backbone, even though people may have downloaded an RT-DETR checkpoint from the hub.

This is only true if they're using a pretrained backbone which they haven't already downloaded. If they set this to a backbone that's within transformers, they'd still have to download these weights.